src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -1294,7 +1298,9 @@ tool_extended!(Miri {
     path: "src/tools/miri",
     tool_name: "miri",
     stable: false,
-    add_bins_to_sysroot: ["miri"]
+    add_bins_to_sysroot: ["miri"],
+    // Avoid costly rebuilds by always including the tests.
```

Comment:
```suggestion
    // Always compile also tests when building miri. Otherwise feature unification can cause rebuilds between building and testing miri.
```